bin/public_interface.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""Extract/compare public interfaces."""
```

Comment:
Should have a blurb about why this exists in the first place. Maybe here.

Something like "transform library hasn't had a publicly defined interface, so we assume anything not considered private by convention (see https://peps.python.org/pep-0008/#descriptive-naming-styles) is public and needs to maintain backward compatibility; this helps automate such checks"

Reply:
added something